app/views/govuk_publishing_components/components/_accordion.html.erb

```diff
@@ -0,0 +1,34 @@
+<%
+  id ||= "default-id-#{SecureRandom.hex(4)}"
+  additional_classes ||= []
```

Comment:
What's the requirement for this? I'm looking into some other components that do this and I believe it's against our component model - it could encourage developers to add glue code to apps, which we want to avoid.

Reply:
What do you mean by 'this'? The `id` or `additional_classes`?

Reply:
The `additional_classes` line.

Reply:
The accordion works in a nice and lovely fashion when given a container that's wide - but is inflexible when not in the ideal layout.

For example - in my particular use case this is the Govspeak guidance is next to the text editor in the content publisher; this is the set of `<details>` that is going to be replaced with an accordion:

> <img width="426" alt="Partial screenshot of the content publisher interface, showing a narrow column containing the Govspeak guidance." src="https://user-images.githubusercontent.com/1732331/51482669-f71dd600-1d8e-11e9-9c93-44066fd0f70f.png">

This is why I'd like to be able to extend the styling of the accordion to make it work when narrower. I think it's better to make it part of the component rather than pushing developers to wrap components in an extra `div` to override the styles.

A modifier could be used instead of adding extra classes - so you'd then have `govuk-accordion govuk-accordion--smaller-width-please` for example. 

Prior art for passing through classes exists in a couple of other components - for example, warning text and error message- though this is not documented for these components.

Reply:
Thanks for the discussion on this yesterday @andysellick - I feel I understand more about the wider context now. 

As we talked about, I'm adding a parameter / flag which will trigger the modifier class, and I'll see about the appetite for feeding this modifier back to the component in GOV.UK Frontend.